### ISSUE TITLE
Mixed content errors on ghost one click deploy

### ISSUE THREAD

**Issue by ravikiranp123** [caprover/caprover#531]
**IMPORTANT:** This is a bug report. If you are having problem with deploying a particular app use the deployment issue type.

> If you are having problems with deploying a particular app using a customized dockerfile - please change your questions to Dockerfile and post it where appropriate such as StackOverflow. This is a Docker question, not a CapRover specific issue.

--------------------------------------------------------------------------

What is the problem?

The one click deploy for ghost gives mixed content error on the admin page and on any post/page images are uploaded. It references http version of the same site even though https and force https are enabled. I have tried setting ghost protocol to https and the port to 443 manually in settings. That did not help though. 

Steps to reproduce the problem:

1. Create a standard ghost app from one click deploy
2. Optionally use 'latest' as version.
3. Enable https and force redirect to it.
4. Open admin page and the scripts try to load from http rather than https resulting in this bug.
5. Optionally upload some image. Say as favicon. This will give you mixed content errors as well.

Answers to the following questions where applicable:
- Your OS and version?
Ubuntu 18.04
- RAM?
1GB

EDIT: Should this go to issues on one click apps repo?

**Comment by githubsaturn** [comment]
Are you using the correct protocol? here:
https://github.com/caprover/one-click-apps/blob/master/public/v2/apps/ghost.json#L110

**Comment by ravikiranp123** [comment]
As mentioned in the original post, yes, I am using the correct https protocol. The issue seems to crop up on the admin page only though. What else would you recommend me to check?

**Comment by githubsaturn** [comment]
If you're using all params that you should be using, then the problem probably lies on ghost itself. 

Find out what resources are being tried to be loaded via http. And open an issue on ghost github page.

**Comment by ravikiranp123** [comment]
All right. I figured it out. I was having issues with openssl if I used https when deploying app. So I used http. This setup admin to use http and was rather hard coded in config and wouldn't update even if the environment variable was changed later. 

There might be better ways to solve it, but this is the fix I found.
The fix is simple.
1. Just cd to '/var/lib/docker/captain--<app name>-data/_data/ghost
2. Edit config.production.json
3. Change your URL to use https.

This should solve it.

I was having this issue on oracle cloud instance running ubuntu 18.04. Hope this helps someone.
